packages/esm-patient-conditions-app/src/conditions/utils.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Enables a comparison of arbitrary values with support for undefined/null.
+ * Requires the `<` and `>` operators to return something reasonable for the provided values.
+ */
+export function compare<T>(x?: T, y?: T) {
```

Comment:
You didn't consider the case for `--`

Reply:
Let me check that, but I was handling it as an empty string.

Reply:
@vasharma05 actually this compare function does not accept the `emdash` values, but I am handling such cases as undefined. Do you have some pointers on how you would want the `emdash` to be handled?

Reply:
The `parseDate` will return a date string and that fails the `typescript` flags it as not being a number.

Reply:
`parseDate()` returns a JS Date object...

Reply:
So:

```ts
  if (x === y || !x && !y) {
    return 0;
  } else if (!x) {
    return -1;
  } else if (!y) {
    return 1;
  } else {
    const xDate = parseDate(x);
    const yDate = parseDate(y);
    
    if (xDate === yDate) {
       return 0;
    }

    return xDate < yDate ? -1 : 1;
  }
```

Reply:
Thanks @ibacher

Reply:
Ohh okay, I remembered `parseDate` returned millseconds, but forgot about the JS Date object.
Thanks Ian for the correction!